File: chaerim/Programmers/Level_2/Lv2_기능개발.py
def solution(progresses, speeds):
    answer = []
    deadline = []

    for progress, speed in zip(progresses, speeds):
        remaining_progress = 100 - progress
        day = remaining_progress // speed
        if remaining_progress%speed != 0:
            day += 1
        deadline.append(day)

    count = 1
    prev_day = deadline[0]
    for day in deadline[1:]:
        if day <= prev_day:
            count += 1
        else:
            answer.append(count)
            count = 1
            prev_day = day
    answer.append(count)

    return answer

File: seongwon/Programmers/Lv2_기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> arr;

    for (int i = 0; i < progresses.size(); i++) {
        int days = (100 - progresses[i] + speeds[i] - 1) / speeds[i]; // 며칠걸리는지 arr에 넣음
        arr.push_back(days);
    }

    int max = arr[0];
    int cnt = 0;
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] <= max) {
            cnt++;
        } else {
            answer.push_back(cnt);
            max = arr[i];
            cnt = 1;
        }
    }
    answer.push_back(cnt);

    return answer;
}
